### DISCUSSION TITLE
Optional limit parameter ?

### DISCUSSION THREAD

**Discussion by uzair004** [sequelize/sequelize#14285] [Help & Questions] [answered]
Is there any way to pass the limit but with optional values, e.g we want to use the same endpoint multiple times but expect different amounts of data from different calls.

so if the user passes the limit = 3  it should be passed to query,
what if the user wants to get all data?
there should be a way of passing limit = "something that returns all values" like a default value.

**Comment by ephys** [comment, marked as answer]
I don't think I understand your question, you can set `limit` to a different value based on the user request

**Reply by uzair004** [reply]
@ephys thanks for the fast reply, basically I want to pass the limit as an optional parameter to find.
e.g 
`api/orders?limit=10`
`api/orders?limit=5`

`api/orders`  (no limit provided)

so passing limit: {} or limit: 0, or limit: ''  should return all data without any limit

basically limit would be a value provided by the user in URL query, if provided as null, empty (other than a number) passing it to `Model.findAll()` should return all data.

e.g
`Model.findAll({where: {id: orderId, limit: 30}})`
`Model.findAll({where: {id: orderId, limit:{}})` should return all data

**Reply by ephys** [reply]
Ah I see, the answer is this I believe:

`Model.findAll({ where: { id: orderId, limit: null } })`

Sequelize will treat `null` and `undefined` as "no limit"

**Reply by uzair004** [reply]
it worked, thanks. 
Appreciate it :)
